conda/cli/main_create.py

```diff
@@ -100,14 +98,25 @@ def configure_parser(sub_parsers: _SubParsersAction, **kwargs) -> ArgumentParser
 
 @notices
 def execute(args: Namespace, parser: ArgumentParser) -> int:
+    import os
+    from tempfile import mktemp
+
     from ..base.context import context
     from ..common.path import paths_equal
-    from ..exceptions import CondaValueError
+    from ..exceptions import ArgumentError, CondaValueError
     from ..gateways.disk.delete import rm_rf
     from ..gateways.disk.test import is_conda_environment
     from .common import confirm_yn
     from .install import install
 
+    if not args.name and not args.prefix:
+        if context.dry_run:
+            context._argparse_args.prefix = os.path.join(mktemp(), "dry-run-create-env")
```

Comment:
```suggestion
            args.prefix = os.path.join(mktemp(), "dry-run-create-env")
            context.__init__(argparse_args=args)
```

I think this is slightly better. No need to access a protected property when we can simply re-initialize the object. Performance penalty should be negligible.

Also, Could we make  `"dry-run-create-env"` a constant and define it in `conda.base.constants`. Not necessary, but would help keep the code a little more organized and provide a central place to look up this value in case it is ever needed.

Reply:
Are we sure we are not overriding some stuff from `os.environ` with that call? 🤔 I always get nervous when I reinitialize the context outside of the tests 😬 

+1 to the 2nd suggestion.

Reply:
Applied your suggestions. The new tests seem to be passing locally.

Reply:
No reason to worry about overriding stuff from `os.environ`. I double checked the code. We just need to make sure we initialize it like we do in `conda.cli.main.main_subshell` by pass the argparse args:

- https://github.com/conda/conda/blob/a261ade3d0a276a214bff314e54c9a9dbb8c282e/conda/cli/main.py#L72